src/gui/MainWindow.cpp

```diff
@@ -407,13 +411,44 @@ MainWindow::MainWindow()
         m_ui->globalMessageWidget->showMessage(
             tr("Access error for config file %1").arg(config()->getFileName()), MessageWidget::Error);
     }
-
+#ifdef WITH_XC_HTTP
+    if (config()->get("Http/Enabled", false).toBool() && !config()->get("Http/DeprecationNoticeShown", false).toBool()) {
+        // show message after tab widget dismissed all messages
+        connect(m_ui->tabWidget, SIGNAL(messageDismissGlobal()), this, SLOT(showKeePassHTTPDeprecationNotice()));
+    }
+#endif
 }
 
 MainWindow::~MainWindow()
 {
 }
 
+void MainWindow::showKeePassHTTPDeprecationNotice()
+{
+    displayGlobalMessage(tr("<p>It looks like you are using KeePassHTTP for browser integration.<br>"
+                                "This feature has been deprecated and will be removed in the future.<br>"
+                                "Please switch to keepassxc-browser instead! For help with migration,<br>"
+                                "visit our <a href=\"https://keepassxc.org/docs/keepassxc-browser-migration\">"
+                                "keepassxc-browser migration guide</a>.</p>"),
+                         MessageWidget::Warning, true, -1);
+
+    config()->set("Http/DeprecationNoticeShown", true);
```

Comment:
Change this to a counter, we will want to show it at least three times. You could write "Final Notice" on the third showing if you want.